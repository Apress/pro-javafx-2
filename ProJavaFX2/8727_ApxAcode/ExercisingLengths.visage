/*
 * @test
 * @run
 */

var onePixel = 1px;
var onePoint = 1pt;
var onePica = 1pc;
var oneDensityIndependentPixel = 1dp;
var oneMillimeter = 1mm;
var oneCentimeter = 1cm;
var oneInch = 1in;
var oneScaleIndependentPixel = 1sp;
var oneEm = 1em;
var onePercent = 1%;

println("onePixel = {%#s onePixel}");
println("onePoint = {%#s onePoint}");
println("onePica = {%#s onePica}");
println("oneDensityIndependentPixel = {oneDensityIndependentPixel}");
println("oneMillimeter = {%#s oneMillimeter}");
println("oneCentimeter = {%#s oneCentimeter}");
println("oneInch = {%#s oneInch}");
println("oneScaleIndependentPixel = {oneScaleIndependentPixel}");
println("oneEm = {oneEm}");
println("onePercent = {onePercent}");

println("3mm + 2mm = {%#s 3mm + 2mm}");
println("5mm - 2mm = {%#s 5mm - 2mm}");
println("3mm * 2 = {%#s 3mm * 2}");
println("2 * 3mm = {%#s 2 * 3mm}");
println("2cm * 2.5 = {%#s 2cm * 2.5}");
println("2.5 * 2cm = {%#s 2.5 * 2cm}");
println("3mm / 2 = {%#s 3mm / 2}");
println("2.5cm / 2.5 = {%#s 2.5cm / 2.5}");
println("2.5mm / 5.0mm = {2.5mm/5.0mm}");

println("oneInch.toCentimeters() = {oneInch.toCentimeters()}");
println("oneInch.toMillimeters() = {oneInch.toMillimeters()}");
println("oneInch.toDensityIndependentPixels() = {oneInch.toDensityIndependentPixels()}");
println("onePica.toPoints() = {onePica.toPoints()}");
println("onePoint.toDensityIndependentPixels() = {onePoint.toDensityIndependentPixels()}");

var tenInches = Length.valueOf(10, visage.lang.LengthUnit.INCH);
println("tenInches = {%#s tenInches}");
