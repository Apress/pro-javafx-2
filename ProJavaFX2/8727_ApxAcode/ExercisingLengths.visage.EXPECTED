onePixel = 1px
onePoint = 1pt
onePica = 1pc
oneDensityIndependentPixel = 1dp
oneMillimeter = 1mm
oneCentimeter = 1cm
oneInch = 1in
oneScaleIndependentPixel = 1sp
oneEm = 1em
onePercent = 1%
3mm + 2mm = 5mm
5mm - 2mm = 3mm
3mm * 2 = 6mm
2 * 3mm = 6mm
2cm * 2.5 = 5cm
2.5 * 2cm = 5cm
3mm / 2 = 1.5mm
2.5cm / 2.5 = 1cm
2.5mm / 5.0mm = 0.5
oneInch.toCentimeters() = 2.5399999618530273
oneInch.toMillimeters() = 25.399999618530273
oneInch.toDensityIndependentPixels() = 96.0
onePica.toPoints() = 12.0
onePoint.toDensityIndependentPixels() = 1.3333333333333333
tenInches = 10in
